hawkular-inventory-impl-tinkerpop-parent/hawkular-inventory-impl-tinkerpop/src/main/java/org/hawkular/inventory/impl/tinkerpop/FilterVisitor.java

```diff
@@ -85,7 +90,7 @@ public void visit(GraphTraversal<?, ?> query, Related related, QueryTranslationS
                 if (null != related.getRelationshipName()) {
                     state.setInEdges(true);
                     state.setComingFrom(Direction.IN);
-                    query.inE(related.getRelationshipName());
+                    query.inE(related.getRelationshipName()).discriminate(discriminator);
```

Comment:
Isn't it possible to "discriminate" the query globally once for all (and then, the discriminator would need to be propagated in the subsequent traversal). It would be safer than discriminating each case one by one, no?
Of course it would mean that the discriminator cannot be changed during a traversal, but if I understand the concept it's a kind of global setting, no?